<?xml version="1.0" encoding="UTF-8" ?>
<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom" xmlns:dc="http://purl.org/dc/elements/1.1/">
  <channel>
    <title>Stackstorm-icinga2 - Exception from data gathered by Sensor</title>
    <link>https://forum.stackstorm.com/t/stackstorm-icinga2-exception-from-data-gathered-by-sensor/1422</link>
    <description>Hello All,

I&#39;m attempting to setup https://github.com/StackStorm-Exchange/stackstorm-icinga2 and after setting up a rule in the Stackstorm UI, I noticed it would never get enforced.

Running the sensor with &quot;/opt/stackstorm/st2/bin/st2sensorcontainer --config-file=/etc/st2/st2.conf --sensor-ref=icinga2.Icinga2StateChangeSensor&quot; as root reveals that many of the detected state changes are causing exceptions in the sensor and thus are not dispatching the trigger. Example state changes causing exceptions below:

Exception Example 1:

    2020-09-17 10:20:52,526 INFO [-] Connecting to event stream API.
    2020-09-17 10:20:52,718 INFO [-] Processing event: {&quot;check_result&quot;:{&quot;active&quot;:true,&quot;check_source&quot;:&quot;###HOSTNAME###&quot;,&quot;command&quot;:[&quot;C:\Program Files\ICINGA2\/sbin/check_load.exe&quot;,&quot;-c&quot;,&quot;95&quot;,&quot;-w&quot;,&quot;90&quot;],&quot;execution_end&quot;:1600363364.508,&quot;execution_start&quot;:1600363362.94,&quot;exit_status&quot;:2.0,&quot;output&quot;:&quot;LOAD CRITICAL 97.28% &quot;,&quot;performance_data&quot;:[&quot;load=97.28%;90;95;0;100&quot;],&quot;schedule_end&quot;:1600363364.512,&quot;schedule_start&quot;:1600363364.512,&quot;state&quot;:2.0,&quot;ttl&quot;:0.0,&quot;type&quot;:&quot;CheckResult&quot;,&quot;vars_after&quot;:{&quot;attempt&quot;:1.0,&quot;reachable&quot;:true,&quot;state&quot;:2.0,&quot;state_type&quot;:0.0},&quot;vars_before&quot;:{&quot;attempt&quot;:1.0,&quot;reachable&quot;:true,&quot;state&quot;:0.0,&quot;state_type&quot;:1.0}},&quot;host&quot;:&quot;###HOSTNAME###&quot;,&quot;service&quot;:&quot;Windows Load&quot;,&quot;state&quot;:2.0,&quot;state_type&quot;:0.0,&quot;timestamp&quot;:1600363363.52031,&quot;type&quot;:&quot;StateChange&quot;}
    2020-09-17 10:20:52,719 ERROR [-] Icinga2StateChangeSensor Exception Invalid \escape: line 1 column 92 (char 91)
    Traceback (most recent call last):
      File &quot;/opt/stackstorm/packs/icinga2/sensors/icinga2_state_change_events.py&quot;, line 66, in run
        self.process_event(event)
      File &quot;/opt/stackstorm/packs/icinga2/sensors/icinga2_state_change_events.py&quot;, line 76, in process_event
        event = json.loads(event)
      File &quot;/usr/lib64/python2.7/json/__init__.py&quot;, line 338, in loads
        return _default_decoder.decode(s)
      File &quot;/usr/lib64/python2.7/json/decoder.py&quot;, line 366, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File &quot;/usr/lib64/python2.7/json/decoder.py&quot;, line 382, in raw_decode
        obj, end = self.scan_once(s, idx)
    ValueError: Invalid \escape: line 1 column 92 (char 91)
    2020-09-17 10:20:57,721 INFO [-] Connecting to event stream API.


Exception Example 2:

    2020-09-17 10:29:03,692 INFO [-] Connecting to event stream API.
    2020-09-17 10:29:29,318 INFO [-] Processing event: {&quot;check_result&quot;:{&quot;active&quot;:true,&quot;check_source&quot;:&quot;###HOSTNAME###&quot;,&quot;command&quot;:[&quot;C:\Windows\system32\WindowsPowerShell\v1.0\powershell.exe&quot;,&quot;-command&quot;,&quot;C:\Program` Files\ICINGA2\sbin\JobStatus.ps1&quot;,&quot;-Username&quot;,&quot;###USERNAME###&quot;,&quot;-Password&quot;,&quot;###PASSWORD###&quot;,&quot;; exit&quot;,&quot;$LASTEXITCODE&quot;],&quot;execution_end&quot;:1600363880.88,&quot;execution_start&quot;:1600363878.467,&quot;exit_status&quot;:0.0,&quot;output&quot;:&quot;OK: All Jobs have an EndTime Entry. &quot;,&quot;performance_data&quot;:[&quot;EndTimeNullCount=0&quot;],&quot;schedule_end&quot;:1600363880.88,&quot;schedule_start&quot;:1600363880.88,&quot;state&quot;:0.0,&quot;ttl&quot;:0.0,&quot;type&quot;:&quot;CheckResult&quot;,&quot;vars_after&quot;:{&quot;attempt&quot;:1.0,&quot;reachable&quot;:true,&quot;state&quot;:0.0,&quot;state_type&quot;:1.0},&quot;vars_before&quot;:{&quot;attempt&quot;:2.0,&quot;reachable&quot;:true,&quot;state&quot;:2.0,&quot;state_type&quot;:0.0}},&quot;host&quot;:&quot;###HOSTNAME###&quot;,&quot;service&quot;:&quot;Job Monitor&quot;,&quot;state&quot;:0.0,&quot;state_type&quot;:1.0,&quot;timestamp&quot;:1600363880.058318,&quot;type&quot;:&quot;StateChange&quot;}
    2020-09-17 10:29:29,320 ERROR [-] Icinga2StateChangeSensor Exception Invalid \escape: line 1 column 89 (char 88)
    Traceback (most recent call last):
      File &quot;/opt/stackstorm/packs/icinga2/sensors/icinga2_state_change_events.py&quot;, line 66, in run
        self.process_event(event)
      File &quot;/opt/stackstorm/packs/icinga2/sensors/icinga2_state_change_events.py&quot;, line 76, in process_event
        event = json.loads(event)
      File &quot;/usr/lib64/python2.7/json/__init__.py&quot;, line 338, in loads
        return _default_decoder.decode(s)
      File &quot;/usr/lib64/python2.7/json/decoder.py&quot;, line 366, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File &quot;/usr/lib64/python2.7/json/decoder.py&quot;, line 382, in raw_decode
        obj, end = self.scan_once(s, idx)
    ValueError: Invalid \escape: line 1 column 89 (char 88)
    2020-09-17 10:29:34,322 INFO [-] Connecting to event stream API.

Has anyone encountered this before and been able to resolve/work around this?</description>
    
    <lastBuildDate>Thu, 17 Sep 2020 18:14:50 +0000</lastBuildDate>
    <category>Pack Management</category>
    <atom:link href="https://forum.stackstorm.com/t/stackstorm-icinga2-exception-from-data-gathered-by-sensor/1422.rss" rel="self" type="application/rss+xml" />
      <item>
        <title>Stackstorm-icinga2 - Exception from data gathered by Sensor</title>
        <dc:creator><![CDATA[Carlos]]></dc:creator>
        <description><![CDATA[
            <p>Looking at the event payload I see this <code>"C:\Program Files\ICINGA2\/sbin/check_load.exe"</code>.</p>
<p>It looks like the string hasn’t been escaped properly.  It’s either a bug in the sensor code or a bug in the underlying  module being used</p>
          <p><a href="https://forum.stackstorm.com/t/stackstorm-icinga2-exception-from-data-gathered-by-sensor/1422/2">Read full topic</a></p>
        ]]></description>
        <link>https://forum.stackstorm.com/t/stackstorm-icinga2-exception-from-data-gathered-by-sensor/1422/2</link>
        <pubDate>Thu, 17 Sep 2020 18:14:50 +0000</pubDate>
        <guid isPermaLink="false">forum.stackstorm.com-post-1422-2</guid>
        <source url="https://forum.stackstorm.com/t/stackstorm-icinga2-exception-from-data-gathered-by-sensor/1422.rss">Stackstorm-icinga2 - Exception from data gathered by Sensor</source>
      </item>
      <item>
        <title>Stackstorm-icinga2 - Exception from data gathered by Sensor</title>
        <dc:creator><![CDATA[thozook]]></dc:creator>
        <description><![CDATA[
            <p>Hello All,</p>
<p>I’m attempting to setup <a href="https://github.com/StackStorm-Exchange/stackstorm-icinga2" class="inline-onebox" rel="noopener nofollow ugc">GitHub - StackStorm-Exchange/stackstorm-icinga2: Icinga2 Integration pack</a> and after setting up a rule in the Stackstorm UI, I noticed it would never get enforced.</p>
<p>Running the sensor with “/opt/stackstorm/st2/bin/st2sensorcontainer --config-file=/etc/st2/st2.conf --sensor-ref=icinga2.Icinga2StateChangeSensor” as root reveals that many of the detected state changes are causing exceptions in the sensor and thus are not dispatching the trigger. Example state changes causing exceptions below:</p>
<p>Exception Example 1:</p>
<pre><code>2020-09-17 10:20:52,526 INFO [-] Connecting to event stream API.
2020-09-17 10:20:52,718 INFO [-] Processing event: {"check_result":{"active":true,"check_source":"###HOSTNAME###","command":["C:\Program Files\ICINGA2\/sbin/check_load.exe","-c","95","-w","90"],"execution_end":1600363364.508,"execution_start":1600363362.94,"exit_status":2.0,"output":"LOAD CRITICAL 97.28% ","performance_data":["load=97.28%;90;95;0;100"],"schedule_end":1600363364.512,"schedule_start":1600363364.512,"state":2.0,"ttl":0.0,"type":"CheckResult","vars_after":{"attempt":1.0,"reachable":true,"state":2.0,"state_type":0.0},"vars_before":{"attempt":1.0,"reachable":true,"state":0.0,"state_type":1.0}},"host":"###HOSTNAME###","service":"Windows Load","state":2.0,"state_type":0.0,"timestamp":1600363363.52031,"type":"StateChange"}
2020-09-17 10:20:52,719 ERROR [-] Icinga2StateChangeSensor Exception Invalid \escape: line 1 column 92 (char 91)
Traceback (most recent call last):
  File "/opt/stackstorm/packs/icinga2/sensors/icinga2_state_change_events.py", line 66, in run
    self.process_event(event)
  File "/opt/stackstorm/packs/icinga2/sensors/icinga2_state_change_events.py", line 76, in process_event
    event = json.loads(event)
  File "/usr/lib64/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python2.7/json/decoder.py", line 382, in raw_decode
    obj, end = self.scan_once(s, idx)
ValueError: Invalid \escape: line 1 column 92 (char 91)
2020-09-17 10:20:57,721 INFO [-] Connecting to event stream API.
</code></pre>
<p>Exception Example 2:</p>
<pre><code>2020-09-17 10:29:03,692 INFO [-] Connecting to event stream API.
2020-09-17 10:29:29,318 INFO [-] Processing event: {"check_result":{"active":true,"check_source":"###HOSTNAME###","command":["C:\Windows\system32\WindowsPowerShell\v1.0\powershell.exe","-command","C:\Program` Files\ICINGA2\sbin\JobStatus.ps1","-Username","###USERNAME###","-Password","###PASSWORD###","; exit","$LASTEXITCODE"],"execution_end":1600363880.88,"execution_start":1600363878.467,"exit_status":0.0,"output":"OK: All Jobs have an EndTime Entry. ","performance_data":["EndTimeNullCount=0"],"schedule_end":1600363880.88,"schedule_start":1600363880.88,"state":0.0,"ttl":0.0,"type":"CheckResult","vars_after":{"attempt":1.0,"reachable":true,"state":0.0,"state_type":1.0},"vars_before":{"attempt":2.0,"reachable":true,"state":2.0,"state_type":0.0}},"host":"###HOSTNAME###","service":"Job Monitor","state":0.0,"state_type":1.0,"timestamp":1600363880.058318,"type":"StateChange"}
2020-09-17 10:29:29,320 ERROR [-] Icinga2StateChangeSensor Exception Invalid \escape: line 1 column 89 (char 88)
Traceback (most recent call last):
  File "/opt/stackstorm/packs/icinga2/sensors/icinga2_state_change_events.py", line 66, in run
    self.process_event(event)
  File "/opt/stackstorm/packs/icinga2/sensors/icinga2_state_change_events.py", line 76, in process_event
    event = json.loads(event)
  File "/usr/lib64/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python2.7/json/decoder.py", line 382, in raw_decode
    obj, end = self.scan_once(s, idx)
ValueError: Invalid \escape: line 1 column 89 (char 88)
2020-09-17 10:29:34,322 INFO [-] Connecting to event stream API.
</code></pre>
<p>Has anyone encountered this before and been able to resolve/work around this?</p>
          <p><a href="https://forum.stackstorm.com/t/stackstorm-icinga2-exception-from-data-gathered-by-sensor/1422/1">Read full topic</a></p>
        ]]></description>
        <link>https://forum.stackstorm.com/t/stackstorm-icinga2-exception-from-data-gathered-by-sensor/1422/1</link>
        <pubDate>Thu, 17 Sep 2020 18:09:02 +0000</pubDate>
        <guid isPermaLink="false">forum.stackstorm.com-post-1422-1</guid>
        <source url="https://forum.stackstorm.com/t/stackstorm-icinga2-exception-from-data-gathered-by-sensor/1422.rss">Stackstorm-icinga2 - Exception from data gathered by Sensor</source>
      </item>
  </channel>
</rss>
